src/Stratis.Bitcoin.Features.PoA/Voting/VotingManager.cs

```diff
@@ -276,9 +276,14 @@ public List<IFederationMember> GetFederationFromExecutedPolls()
                     IFederationMember federationMember = ((PoAConsensusFactory)(this.network.Consensus.ConsensusFactory)).DeserializeFederationMember(poll.VotingData.Data);
 
                     if (poll.VotingData.Key == VoteKey.AddFederationMember)
-                        federation.Add(federationMember);
+                    {
+                        if (!federation.Any(m => m is CollateralFederationMember colMember && federationMember is CollateralFederationMember colMember2 && colMember.CollateralMainchainAddress == colMember2.CollateralMainchainAddress))
```

Comment:
Shouldnt the `federationMember is CollateralFederationMember colMember2` part here be outside of the `Any` ?

It probs amounts to the same but it reads better and also the type doesnt happen for every member in the federation?

Reply:
Or is this acting as a compare then?

Reply:
Good spot